pkg/apply/apply.go

```diff
@@ -658,6 +669,7 @@ func (t *TopicApplier) updatePartitionsHelper(
 		"Update complete; new partition status:\n%s",
 		admin.FormatTopicPartitions(topicInfo.Partitions, t.brokers),
 	)
+	changes.NumPartitions = partitionCountChanges
```

Comment:
I think you want to do this at line 662 and ensure you always return the result. 
The change to the topic is performed at line 658. So it may succeed while the `GetTopic` at line 663 may fail. 
Then the partitions would be updated without the user know.